### PR TITLE
Updated Angular App To Angular 10

### PR DESCRIPTION
**Pull request by sentryadam0000345**
@thinkocapo

Updated Angular app to use latest version of Angular, Angular 10, updated minor changes such as passing in a dynamic environment release variable.  This involved updating our predeploy package.json script as well as adding a new config-env.ts file.

TypeError: https://sentry.io/organizations/testorg-az/issues/1839780499/?project=1190123&query=is%3Aunresolved

URIError: https://sentry.io/organizations/testorg-az/issues/1839780491/?project=1190123&query=is%3Aunresolved

SyntaxError: https://sentry.io/organizations/testorg-az/issues/1841114418/?project=1190123&query=is%3Aunresolved

RangeError:  https://sentry.io/organizations/testorg-az/issues/1841114413/?project=1190123&query=is%3Aunresolved

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
```
Comment (thinkocapo) [inline src/app/app-routing.module.ts]: angular/router, sweet.

File: Makefile [hunk 2]
```diff
@@ -7,17 +7,21 @@ SENTRY_PROJECT=ng-demo
 VERSION=`sentry-cli releases propose-version`
 PREFIX=dist
 
+
```
Comment (ndmanvar) [inline Makefile]: remove

File: e2e/src/app.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { AppPage } from './app.po';
```
Comment (ndmanvar) [inline e2e/src/app.e2e-spec.ts]: remove this test

File: package.json [hunk 4]
```diff
@@ -1,49 +1,56 @@
...
-    "karma": "2.0.0",
-    "karma-chrome-launcher": "^2.0.0",
-    "karma-cli": "^1.0.1",
-    "karma-jasmine": "^1.1.2",
-    "karma-remap-istanbul": "^0.6.0",
-    "protractor": "^5.3.2",
-    "ts-node": "6.0.0",
-    "tslint": "^5.12.1",
-    "typescript": "^2.8.4"
+    "@angular-builders/custom-webpack": "^10.0.0",
```
Comment (ndmanvar) [inline package.json]: are these necessary?

File: src/app/app.component.ts [hunk 5]
```diff
@@ -49,4 +49,4 @@ export class AppComponent {
   }
   // ERRORS (end)
 
-}
+}
```
Comment (ndmanvar) [inline src/app/app.component.ts]: always have newline at end

File: src/app/app.module.ts [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
 
 import * as Sentry from "@sentry/browser";
 
 Sentry.init({
-  dsn: "https://60443a2c0a0c45599b1387046605bc88@sentry.io/269722",
-  release: "b6d7c8d5e2d0ca00c65090c8761ca07cefd95f12"
-});
+  dsn: "https://6dd6b79934654822b631951be27ec0e1@o87286.ingest.sentry.io/1190123",
+  release: environment.release
+  //TODO dynamic release, pass in via environment variable, pass in somehow, not static
```
Comment (ndmanvar) [inline src/app/app.module.ts]: remove comment

File: src/environments/environment.ts [hunk 7]
```diff
@@ -1,8 +1,4 @@
-// The file contents for the current environment will overwrite these during build.
-// The build system defaults to the dev environment which uses `environment.ts`, but if you do
-// `ng build --env=prod` then `environment.prod.ts` will be used instead.
-// The list of which env maps to which file can be found in `angular-cli.json`.
-
 export const environment = {
-  production: false
-};
+    release: '54e59e0adaa690fcd5ae9808a70d6e0bce2fe165'
```
Comment (ndmanvar) [inline src/environments/environment.ts]: `<TO_BE_REPLACED>`


in environments-prod.ts put `// auto-generated by xyz script`

File: src/main.ts [hunk 8]
```diff
@@ -1,10 +1,12 @@
...
-import { environment } from './environments/environment';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+
 import { AppModule } from './app/app.module';
+import { environment } from './environments/environment';
 
-if (environment.production) {
-  enableProdMode();
-}
+//if (environment.production) {
```
Comment (ndmanvar) [inline src/main.ts]: rmeove unnecessary comment

File: touch.txt [hunk 9]
```diff
@@ -0,0 +1 @@
+est
```
Comment (ndmanvar) [inline touch.txt]: remove this file

File: tslint.json [hunk 10]
```diff
@@ -40,77 +63,90 @@
...
+    "no-outputs-metadata-property": true,
+    "template-banana-in-box": true,
+    "template-no-negated-async": true,
+    "use-lifecycle-interface": true,
+    "use-pipe-transform-interface": true
+  },
+  "rulesDirectory": [
+    "codelyzer"
+  ]
+}
```
Comment (ndmanvar) [inline tslint.json]: add new lines plesase

File: Makefile [hunk 11]
```diff
@@ -7,17 +7,21 @@ SENTRY_PROJECT=ng-demo
...
-	sentry-cli releases -o $(SENTRY_ORG) -p $(SENTRY_PROJECT) set-commits --auto $(VERSION)
+	-sentry-cli releases -o $(SENTRY_ORG) -p $(SENTRY_PROJECT) set-commits --local $(VERSION) 
 
 upload_sourcemaps:
 	sentry-cli releases -o $(SENTRY_ORG) -p $(SENTRY_PROJECT) files \
 		$(VERSION) upload-sourcemaps --url-prefix "~/" --rewrite --validate $(PREFIX)
 
-reference_release:
-	sed -i -e "s/release: .*/\release: \"${VERSION}\"/g" src/app/app.module.ts
+create_env:
```
Comment (thinkocapo) [inline Makefile:24]: can you notate in the readme how this command is intended to be used?

Does it belong under Setup (i.e. command you should only have to run once) or Run (need to run it every time you want to run this angular app)

File: .env [hunk 12]
```diff
@@ -0,0 +1 @@
+version=54e59e0adaa690fcd5ae9808a70d6e0bce2fe165
```
Comment (thinkocapo) [inline .env:1]: notate in the README that the developer must create a .env file, unless this is auto generated for you somehow.

typically a .env's are ignored from source control, but if it's just taking a version (release version?) then that's okay here.

File: src/app/app.module.ts (hunk 6 above)
Comment (thinkocapo) [inline src/app/app.module.ts]: remove this TODO if it's no longer needed

**Comment by thinkocapo** [comment]
I see a lot of Tests got committed. Did you use an angular app generator or were you just incrementing the version of the angular npm modules? just wondering

**Review by ndmanvar — CHANGES_REQUESTED** [review]

**Comment by ndmanvar** [comment]
@sentryadam0000345  can you change the title of the PR also?

File: Makefile (hunk 11 above)
Comment (sentryadam0000345) [reply Makefile:24]: @thinkocapo create_env is used as part of predeploy, you never run it directly, when you run deploy, predeploy kicks in, I will note this in the readme

Comment (thinkocapo) [reply Makefile:24]: ok @sentryadam0000345 thanks that makes sense.

File: Makefile (hunk 2 above)
Comment (sentryadam0000345) [reply Makefile]: @ndmanvar I use version and prefix in the Makefile, can't remove this as a result

File: Makefile [hunk 13]
```diff
@@ -7,17 +7,21 @@ SENTRY_PROJECT=ng-demo
 VERSION=`sentry-cli releases propose-version`
```
Comment (ndmanvar) [inline Makefile]: this should reference `./node_modules/.bin/sentry-cli`

File: README.md [hunk 14]
```diff
@@ -14,7 +14,15 @@ https://docs.sentry.io/platforms/javascript/angular/
...
 The following command is defined in [package.json](https://github.com/sentry-demos/angular/blob/master/package.json#L12-L13) and will build, create release, upload source maps and deploy:
 
-`npm run deploy`
+`npm run deploy` 
+
+note: predeploy runs as part of deploy where create_env is used
+
+note: a .env file must be made in order to store the relative variable value provided by the Sentry CLI 
+
+
```
Comment (ndmanvar) [inline README.md:23]: remvoe excess wshitespace

**Review by ndmanvar — CHANGES_REQUESTED** [review]

**Comment by ndmanvar** [comment]
looks good to me. feel free to merge after resolving @sentryadam0000345

File: Makefile (hunk 13 above)
Comment (sentryadam0000345) [reply Makefile]: @ndmanvar why should this be ./node_modules/.bin/sentry-cli?

File: Makefile [hunk 15]
```diff
@@ -9,15 +9,17 @@ PREFIX=dist
...
 setup_release: create_release associate_commits upload_sourcemaps
 
+
 create_release:
 	sentry-cli releases -o $(SENTRY_ORG) new -p $(SENTRY_PROJECT) $(VERSION)
 
+
 associate_commits:
-	sentry-cli releases -o $(SENTRY_ORG) -p $(SENTRY_PROJECT) set-commits --auto $(VERSION)
+	-sentry-cli releases -o $(SENTRY_ORG) -p $(SENTRY_PROJECT) set-commits --local $(VERSION)
```
Comment (ndmanvar) [inline Makefile:18]: let's call sentry-cli via viariable:
```
SENTRY_CLI=./node_modules/.bin/sentry-cli
...
associate_commits:
    $(SENTRY_CLI) releases -o ....
```

**Review by ndmanvar — COMMENTED** [review]
@sentryadam0000345

**Pull request merged**